src/client.rs

```diff
@@ -544,10 +546,12 @@ pub trait CryptoClient: PollClient {
         &mut self,
         raw_key: &[u8],
         location: Location,
+        kind: Kind,
```

Comment:
`key::Kind` is not exposed in the `Client` API currently, I'm not sure it should. To me it's more an internal detail an the `mechanism` enum should be used instead.

Reply:
If so, then surely we can go to the original API, without the needless extension.